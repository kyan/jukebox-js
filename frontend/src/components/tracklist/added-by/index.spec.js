import React from 'react'
import { shallow } from 'enzyme'
import AddedBy from './index'

const firstUser = {
  _id: '123',
  fullname: 'Big Rainbowhead',
  addedAt: '2019-12-17T13:11:37.316Z'
}
const secondUser = {
  _id: '456',
  fullname: 'Big Rainbowhead2',
  addedAt: '2019-12-17T13:11:37.316Z'
}

describe('AddedBy', () => {
  let wrapper

  describe('when no addedBy information provided', () => {
    it('displays a spotify icon', () => {
      wrapper = shallow(<AddedBy />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when addedBy information provided', () => {
    describe('and it is the first play', () => {
      it('displays a current user icon with a popup message', () => {
        wrapper = shallow(<AddedBy users={[firstUser]} />)
        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('and there is play history', () => {
      it('displays a current user icon with a popup message containing previous ', () => {
        wrapper = shallow(<AddedBy users={[firstUser, secondUser]} />)
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})
