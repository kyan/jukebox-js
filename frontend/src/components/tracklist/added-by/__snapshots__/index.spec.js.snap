// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddedBy when addedBy information provided and it is the first play displays a current user icon with a popup message 1`] = `
<Popup
  content="This is the first play!"
  disabled={false}
  eventsEnabled={true}
  offset={0}
  on={
    Array [
      "click",
      "hover",
    ]
  }
  pinned={false}
  position="top left"
  trigger={
    <span
      className="added-by"
    >
      <Icon
        as="i"
        name="user outline"
      />
      Big Rainbowhead
    </span>
  }
/>
`;

exports[`AddedBy when addedBy information provided and there is play history displays a current user icon with a popup message containing previous  1`] = `
<Popup
  content={
    <List>
      <ListItem>
        Big Rainbowhead
         on 
        December 17, 2019
      </ListItem>
    </List>
  }
  disabled={false}
  eventsEnabled={true}
  offset={0}
  on={
    Array [
      "click",
      "hover",
    ]
  }
  pinned={false}
  position="top left"
  trigger={
    <span
      className="added-by"
    >
      <Icon
        as="i"
        name="user outline"
      />
      Big Rainbowhead2
    </span>
  }
/>
`;

exports[`AddedBy when no addedBy information provided displays a spotify icon 1`] = `
<Icon
  as="i"
  name="spotify"
/>
`;
